### PR TITLE
allow collaborators view and edit a shared todo

### LINKED ISSUE #121: Can't view/edit a shared todo
### What's going wrong and not working?

After accepting a shared todo, the invited user can't view and edit the todo as our backend returns a 403.

### Expected outcome

Being able to edit/view a shared todo.

### PR DESCRIPTION
**Pull request by rieckpil**
See #121.

The main issue was that we only checked for the ownership of the todo and did not verify if the user of the to-be-shown/to-be-edited todo is a collaborator.

Added a test for that ☑️

On top of the bug fix, I updated all Gradle versions with `./gradlew wrapper --gradle-version 7.5` as recommended in the [docs](https://docs.gradle.org/current/userguide/gradle_wrapper.html#sec:upgrading_wrapper).

There's no need to update the book as the targeted code parts are not part of our manuscript.

If everything looks fine, you can squash and merge this. Adding "closes #121" to the commit body of the squash will automatically close the corresponding GitHub issue.

### CODE REVIEW

File: application/src/main/java/dev/stratospheric/todoapp/todo/TodoController.java [hunk 1]
```diff
@@ -35,7 +35,7 @@ public String showView(
     Model model
   ) {
 
-    Todo todo = todoService.getOwnedTodo(id, user.getEmail());
+    Todo todo = todoService.getOwnedOrShared(id, user.getEmail());
```
Comment (BjoernKW) [inline application/src/main/java/dev/stratospheric/todoapp/todo/TodoController.java:38]: This method should probably be named `getOwnedOrSharedTodo`

File: application/src/main/java/dev/stratospheric/todoapp/todo/TodoService.java [hunk 2]
```diff
@@ -59,20 +53,40 @@ public void updateTodo(Todo updatedTodo, long id, String ownerEmail) {
     this.todoRepository.save(existingTodo);
   }
 
-  public Todo getOwnedTodo(long id, String ownerEmail) {
+  public void delete(long id, String ownerEmail) {
+    Todo toBeDeletedTodo = getOwnedTodo(id, ownerEmail);
+    this.todoRepository.delete(toBeDeletedTodo);
```
Comment (BjoernKW) [inline application/src/main/java/dev/stratospheric/todoapp/todo/TodoService.java:58]:
```suggestion
    this.todoRepository.delete(getOwnedTodo(id, ownerEmail));
```

**Review by BjoernKW — CHANGES_REQUESTED** [review]
Just two minor change requests. Looks great otherwise! Thanks for the additional test coverage.

File: application/src/main/java/dev/stratospheric/todoapp/todo/TodoController.java (hunk 1 above)
Comment (rieckpil) [reply application/src/main/java/dev/stratospheric/todoapp/todo/TodoController.java]: 🦅 eye - you're right, thanks

File: application/src/main/java/dev/stratospheric/todoapp/todo/TodoService.java (hunk 2 above)
Comment (rieckpil) [reply application/src/main/java/dev/stratospheric/todoapp/todo/TodoService.java]: yup. makes sense

**Pull request merged**